### ISSUE TITLE
International Shipping

### ISSUE THREAD

**Issue by hverdonk** [vsoch/freegenes#107]
Is there a way to ship packages internationally/fill out customs forms through the main site? The "Create Shipment" form doesn't have a field to specify another country.

**Comment by vsoch** [comment]
If you can show me the exact way to do this with the Shippo API, this could be added. @Koeng101 is this in scope of the current work?

**Comment by vsoch** [comment]
@Koeng101 could you please provide what the customs generation (and the various fields they need in the example) would look like? I can work on it if I know this information.

**Comment by vsoch** [comment]
Also just to verify for you @hverdonk - scope of work refers to the original support agreement that was done for Endy Lab. International shipments would need to be scoped to that set of items.

**Comment by hverdonk** [comment]
Ah, my mistake - I was referring to the scope of our project itself, and was unaware that your contract may not cover the whole thing. I'll let Keoni respond.

**Comment by vsoch** [comment]
Yeah no worries! If it's a fairly simple addition then I'd be happy to do it. But looking at customs forms, (at least at face value) it seems like it might be a bit complex. Let me look in greater detail a bit later and  that will also give Keoni some time to respond.

**Comment by Koeng101** [comment]
Hey @vsoch 

I looked into the problem a little, and it seems like it's not *too* difficult, but at the same time, we don't have too many international orders, and requirements may change for how we do international shipments.

As an intermediate solution, would it be possible to manually link an `Order` (in our system) to a shipment that is created manually in Shippo's system? This would make it so we can process international shipments well enough and I would imagine integration of that feature would make everyone happy. Once we really know what we need to ship internationally robustly, then it would be the time to integrate more tightly - but IMO a solution around connecting the two manually seems reasonable for right now.

**Comment by vsoch** [comment]
That's a great idea @Koeng101 ! What we can do is have a form that will accept a Shippo shipment, and then (other than creation) the same operations should be possible.

@hverdonk are you able to create a dummy (or real) international shipment and provide an id so I can develop this?

**Comment by hverdonk** [comment]
Sure! Where would I find the ID?

**Comment by vsoch** [comment]
Hmm I'm not actually sure, I've never used Shippo (via the web interface) before! @Koeng101 do you know off the top of your head?

**Comment by vsoch** [comment]
Another idea is to present some form that lets you choose from ALL shipments that have ever been generated - how would you best be able to identify a shipment? Based on a date and to_address name?

**Comment by vsoch** [comment]
It looks like I can get hairy details if I paginate this:
```
shipments = shippo.Shipment.all(api_key=os.environ.get('SHIPPO_TOKEN'))
```
And then I'd need to present the options in a select with the date and to_address name viewable to the user, and the value being the id. Importing the id would then mean again querying shippo to get the shipment, and then re-populating the order fields (with what I hope is enough / nothing missing!)

So if that works - @hverdonk do you have a name, date that I can use for testing (that is a shipment done with shippo that isn't added to the system, but you'd want to import?) I can work on this tomorrow.

**Comment by hverdonk** [comment]
Unfortunately, I don't have any international shipments that haven't been imported already.

**Comment by vsoch** [comment]
Can you share one that is international, even if imported? If I test locally I won't have it imported, and I can just test adding to my local server (and won't issue any new labels or anything like that!)

**Comment by hverdonk** [comment]
Sure! I shipped one to Dr. Jennifer Molloy at Cambridge on August 14th, 2019.

**Comment by vsoch** [comment]
Perfect, I'll see if I can put something together from that.

**Comment by vsoch** [comment]
Unfortunately I've found no way to use the Shippo API to get an exact transaction and label for a previous order, see https://github.com/vsoch/freegenes/pull/108. This view likely won't be possible to add, but @Koeng101 can take a look.

**Comment by vsoch** [comment]
I've updated the label here to be for a future feature (out of scope of current work) and in the meantime we have added Shippo Import #108 so you can create the international shipments, and still import the label and tracking urls.
